src/main/java/org/opensearch/sdk/ScriptExtension.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.script.ScriptContext;
+import org.opensearch.script.ScriptEngine;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An additional extension point for {@link Extension}s that extends OpenSearch's scripting functionality.
+ *
+ * @opensearch.api
+ */
+public interface ScriptExtension {
```

Comment:
@dbwiddis as we move over more extension points, we should probably group all these interfaces in one directory.

Reply:
@saratvemulapalli already would have done it if I could [think of a good package name](https://martinfowler.com/bliki/TwoHardThings.html).